pulp_smash/tests/docker/cli/test_sync_publish.py

```diff
@@ -1,338 +1,228 @@
-
-
-class InvalidFeedTestCase(_BaseTestCase):
+            repo_id=repo_id,
+            upstream_name=DOCKER_UPSTREAM_NAME.split('/')[-1],  # drop library/
+        ).stdout)
+        self.addCleanup(docker_utils.repo_delete, self.cfg, repo_id)
+        self.verify_proc(docker_utils.repo_sync(self.cfg, repo_id))
+        self.verify_proc(docker_utils.repo_publish(self.cfg, repo_id))
+
+        # Make Crane read the metadata. (Now!)
+        cli.ServiceManager(self.cfg).restart(('httpd',))
+
+        # Get and inspect /crane/repositories.
+        client = self.make_crane_client(self.cfg)
+        repos = client.get('/crane/repositories')
+        self.assertIn(repo_id, repos.keys())
+        with self.subTest():
+            self.assertFalse(repos[repo_id]['protected'])
+        if selectors.bug_is_testable(2723, self.cfg.version):
+            with self.subTest():
+                self.assertTrue(repos[repo_id]['image_ids'])
+            with self.subTest():
+                self.assertTrue(repos[repo_id]['tags'])
+
+
+class InvalidFeedTestCase(utils.BaseAPITestCase):
     """Show Pulp behaves correctly when syncing a repo with an invalid feed."""
 
-    @classmethod
-    def setUpClass(cls):
+    def test_all(self):
         """Create a docker repo with an invalid feed and sync it."""
-        super(InvalidFeedTestCase, cls).setUpClass()
-        docker_utils.repo_create(
-            cls.cfg,
+        repo_id = utils.uuid4()
+        self.assertNotIn('Task Failed', docker_utils.repo_create(
+            self.cfg,
             feed='https://docker.example.com',
-            repo_id=cls.repo_id,
+            repo_id=repo_id,
             upstream_name=DOCKER_UPSTREAM_NAME,
-        )
-        cls.completed_proc = cli.Client(cls.cfg, cli.echo_handler).run(
-            'pulp-admin docker repo sync run --repo-id {}'
-            .format(cls.repo_id).split()
-        )
-
-    def test_return_code(self):
-        """Assert the "sync" command has a non-zero return code."""
-        if selectors.bug_is_untestable(427, self.cfg.version):
-            self.skipTest('https://pulp.plan.io/issues/427')
-        self.assertNotEqual(self.completed_proc.returncode, 0)
-
-    def test_task_succeeded(self):
-        """Assert the phrase "Task Succeeded" is not in stdout."""
-        self.assertNotIn('Task Succeeded', self.completed_proc.stdout)
-
-    def test_task_failed(self):
-        """Assert the phrase "Task Failed" is in stdout."""
-        self.assertIn('Task Failed', self.completed_proc.stdout)
+        ).stdout)
+        self.addCleanup(docker_utils.repo_delete, self.cfg, repo_id)
+        client = cli.Client(self.cfg, cli.echo_handler)
+        proc = client.run((
+            'pulp-admin', 'docker', 'repo', 'sync', 'run', '--repo-id', repo_id
+        ))
+        if selectors.bug_is_testable(427, self.cfg.version):
+            with self.subTest():
+                self.assertNotEqual(proc.returncode, 0)
+        with self.subTest():
```

Comment:
No. `verify_proc()` asserts that a process succeeded. This code asserts that a process failed.